### DISCUSSION TITLE
issue: Reset not reseting on this circumstance

### DISCUSSION THREAD

**Discussion by pschelte** [react-hook-form/react-hook-form#10133] [Q&A] [answered]
### Version Number

7.43.2

### Codesandbox/Expo snack

https://codesandbox.io/s/mystifying-ben-6nxz1f?file=/src/App.js

### Steps to reproduce

1. Fill the label field
2. Choose a color
3. Click on "Validate"
4. see the console log with the form values
5. the form is not reseted to the default values

### Expected behaviour

The form should be reseted with the default values

### What browsers are you seeing the problem on?

Firefox, Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
What do you mean by not resetting and why console.log is before the reset? What are you try to achieve?

**Reply by pschelte** [reply]
@bluebill1049 in fact the console.log was just here to show that the form was working well. and I just wanted to empty the form just after submitting it. In my current whole application I need this to work this way.

**Comment by leapful** [comment, marked as answer]
@pschelte 

I think you are mis-understanding how `Controller` of react-hook-form and also real `input` element work. 

You customize a lot of things inside your `InputField` but forget to prop `value` to `input` element therefore when react-hook-form reset form values, default value will be set on react-hook-form reference but not on `input` element therefore the UI is now out of sync with react-hook-form data.

The following updated demo with setting prop `value` will solve the issue: https://codesandbox.io/s/empty-sun-csriwr?file=/src/InputField.js

**Reply by pschelte** [reply]
@leapful , in fact yes I have made a mix of a controlled and not controlled input by using this defaultValue, I'm surprise that before encountered this issue, all the other input fields of my application were working as expected, but because I was not resetting them.
By the way thanks a lot !
